src/DynamoCore/Configuration/PathManager.cs

```diff
@@ -80,9 +80,16 @@ class PathManager : IPathManager
         private readonly HashSet<string> preloadedLibraries;
         private readonly HashSet<string> extensionsDirectories;
         private readonly HashSet<string> viewExtensionsDirectories;
-        
+
         #endregion
 
+        internal IPreferences Preferences { get; set; }
+
+        private List<string> RootDirectories
```

Comment:
@sharadkjaiswal I tried to use only `PreferenceSettings.CustomPackageFolders` here, but it requires creating and initializing `PreferenceSettings` *before* `PathManager` is created. As far as `PathManager` takes part in initializing `PreferenceSettings`, it requires creating `PathManager` first. So, I don't see any good way to get rid of `rootDirectories`